sympy/core/expr.py

```diff
@@ -146,19 +146,9 @@ def _hashable_content(self):
         return self._args
 
     def __eq__(self, other):
-        if not isinstance(other, Basic):
```

Comment:
Is this method needed at all? Can the Basic method not also work here?

Reply:
Removing it caused failures in the test. Probably because of the check "number *in* an expression" part.

Reply:
Which tests failed?